scripts/basic-workflow-tests.sh

```diff
@@ -283,7 +312,7 @@ test_train() {
         ilab model train "${TRAIN_ARGS[@]}"
     else
         # TODO Only cuda for now
-        TRAIN_ARGS+=("--legacy" "--device=cuda")
+        TRAIN_ARGS+=("--legacy" "--device=cuda" "--data-path" "${DATA_HOME}/instructlab/datasets")
```

Comment:
I may miss something, but I think `train` command already defaults to whatever is passed in the train profile (or in the base config). Both are currently pointing to this directory already. (The a10 train profile is updated to point to this directory in this PR.)

Why is this change needed here?

Reply:
Which profile hardcodes it to a different path than `${DATA_HOME}/instructlab/datasets`? Thought you modify `train-profile-a10.yaml` in this PR to point there already. Is there some other profile that you are handling here?

Reply:
The modifications made to the train profile don't matter anymore since they're not going to be correct in either case. But previously, they were pointing to models which were located at a local directory relative to the caller. This was fine in the past because legacy checkpoints would be stored in the `./checkpoints`. But since we've updated this behavior to be consistent with the rest of the persistent storage system, looking in `./checkpoints` will break.

At the moment, the `--train-profile` flag requires all of the fields to be present so we cannot simply omit them. But we should retain the new changes as they are at least slightly more accurate to how the train profile would behave on a regular system as opposed to the current variation.

Reply:
Thanks for explanation. Indeed, the profile path won't work anyway since it's not under `E2E_TEST_DIR`.